### ISSUE TITLE
Por Onde Começar?

### ISSUE THREAD

**Issue by lucastafarelbs** [react-native-drops/react-native-faq#2]
Quanto de React tem que conhecer para poder começar a estudar React-Native, é um pré-requisito entender React? Por onde começar os estudos?

**Comment by rafaellincoln** [comment]
Olá @lucastafarelbs, essa é uma ótima pergunta, muitas pessoas possuem essa mesma dúvida, então vamos lá.

O React e React Native compartilham a mesma sintase (JSX) e o mesmo fluxo de funcionamento. Então se você entende como o React funciona, será mais simples você entender o React Native e como ele renderiza as view, como ele atualiza os estados e por ai vai. Mas se você está começando agora com React / React Native, nada te impede de ir direto pro React Native e aprender como tudo funciona utilizando o React Native, então o React não é um pré-requisito, mas isso se você conhece React então você estará um passo a frente.

A documentação do React Native é um bom lugar para começar, alem de vários cursos e vídeos no youtube.

**Comment by josenaves** [comment]
Eu não concordo @rafaellincoln . Você não precisa estudar React profundamente para aprender React Native. Porém, precisa estudar o básico de React:

- Component
- Props
- State
- JSX
- render
- Ciclo de vida do componente (componentWillMout, componentDidMount, etc..)

Vale lembrar que em todos os componentes de React Native que fazemos usamos esse import

```
import React, { Component } from 'react';
```

**Comment by josenaves** [comment]
Eu começaria pela documentação do React - [este tutorial vai muito bem](https://reactjs.org/tutorial/tutorial.html)

Depois, tem esse excelente [React Native Express](http://www.reactnativeexpress.com/)

Não deixe se intimidar com Redux, GraphQL. Leva um tempo para aprender isso. Aprenda o básico. 

Ah, se você não estiver totalmente totalmente confortável com ES6, as coisas vão ficar um pouco mais complicadas. Código ES6 é muito mais legível e compacto. Use-o !

**Comment by rafaellincoln** [comment]
Foi justamento o que tentei dizer @josenaves, e como disse o React não é um pré-requisito, você deve entender somente a sintase (JSX) e o fluxo de funcionamento do React / React Native (Component, PureComponent, state, props, flux (redux), render, shouldComponentUpdate...).

Claro que se souber React isso ira te ajudar, pois assim terá familiaridade com o fluxo do React / React Native, mas não é uma obrigatoriedade.

Recomendo também ler o [artigo](http://makeitopen.com/) sobre o F8, onde os engenheiros do Facebook explicam como fizeram o APP da Facebook Developer Conference.

**Comment by lucastafarelbs** [comment]
Pode crer, valeu @rafaellincoln e @josenaves

**Comment by Geissoncr** [comment]
Concordo com as opiniões de vcs de que react e react-native compartilham os mesmo paradigmas de desenvolvimento, mas por serem os mesmos paradigmas é indiferente vc aprender primeiro react ou react-native... Se seu objetivo for mobile já vai pro react-native bola pra frente!

**Comment by kevinbreaker** [comment]
O primeiro passo eu diria que é conhecer o JavaScript e melhor ainda o ES6/ES7, por que isso sim é fundamental. Não há necessidade de saber React pra entrar no React-Native, se você aprender um ajudará a entender o outro,  ambos utilizam JSX e tem o mesmo paradigma.  Eu por exemplo já fui direto para o RN. O que recomendo fazer é olhar o conceito de componentização, caso você já tenha visto o angular ou vuejs será  um pouco mais tranquilo.
